src/plugins/intel_cpu/src/nodes/multinomial.cpp

```diff
@@ -0,0 +1,176 @@
+
+    for (size_t i = 0lu; i < op->get_input_size(); i++) {
+        if (is_type<op::v0::Constant>(op->get_input_node_ptr(i))) {
+            m_const_inputs[i] = true;
+        }
+    }
+}
+
+bool Multinomial::isSupportedOperation(const std::shared_ptr<const ov::Node>& op, std::string& errorMessage) noexcept {
+    try {
+        if (op->get_type_info() != op::v13::Multinomial::get_type_info_static()) {
+            errorMessage = "Only Multinomial operation from the opset13 is supported by the CPU plugin.";
+            return false;
+        }
+    } catch (...) {
+        return false;
+    }
+    return true;
+}
+
+void Multinomial::getSupportedDescriptors() {
+    if (getParentEdges().size() != 2) {
+        THROW_CPU_NODE_ERR("has incorrect number of input edges.");
+    }
+    if (getChildEdges().size() != 1) {
+        THROW_CPU_NODE_ERR("has incorrect number of output edges.");
+    }
+}
+
+void Multinomial::initSupportedPrimitiveDescriptors() {
+    m_probs_precision = getOriginalInputPrecisionAtPort(PROBS_PORT);
+    if (!one_of(m_probs_precision, ov::element::f32, ov::element::f16, ov::element::bf16)) {
+        m_probs_precision = ov::element::f32;
+    }
+
+    m_num_samples_precision = getOriginalInputPrecisionAtPort(NUM_SAMPLES_PORT);
+    if (!one_of(m_num_samples_precision, ov::element::i64, ov::element::i32)) {
+        m_num_samples_precision = ov::element::i32;
+    }
+
+    addSupportedPrimDesc({{LayoutType::ncsp, m_probs_precision, m_const_inputs[PROBS_PORT]},
+                          {LayoutType::ncsp, m_num_samples_precision, m_const_inputs[NUM_SAMPLES_PORT]}},
+                         {{LayoutType::ncsp, m_output_precision}},
+                         ref_any);
+}
+
+std::string Multinomial::getPrimitiveDescriptorType() const {
+    std::string str_type;
+    auto selectedPrimitiveDesc = getSelectedPrimitiveDescriptor();
+
+    impl_desc_type type = impl_desc_type::undef;
+    if (selectedPrimitiveDesc) {
+        type = selectedPrimitiveDesc->getImplementationType();
+    }
+
+    if (type == impl_desc_type::unknown)
+        str_type += "unknown_";
+    if ((type & impl_desc_type::jit) == impl_desc_type::jit)
+        str_type += "jit_";
+    if ((type & impl_desc_type::ref) == impl_desc_type::ref)
+        str_type += "ref_";
+    if ((type & impl_desc_type::avx512) == impl_desc_type::avx512)
+        str_type += "avx512_";
+    if ((type & impl_desc_type::avx2) == impl_desc_type::avx2)
+        str_type += "avx2_";
+    if ((type & impl_desc_type::sse42) == impl_desc_type::sse42)
+        str_type += "sse42_";
+    if ((type & impl_desc_type::any) == impl_desc_type::any)
+        str_type += "any_";
+
+    if (str_type.empty())
+        str_type += "undef_";
+
+    if (selectedPrimitiveDesc) {
+        str_type += m_output_precision.get_type_name();
+    } else {
+        str_type.pop_back();
+    }
+
+    return str_type;
+}
+
+bool Multinomial::needShapeInfer() const {
+    return true;
```

Comment:
Shape inference is not required if batch and second input are constant.

Reply:
Right, added a constraint!